### ISSUE TITLE
Implement APIFlask for automated Swagger docs

### ISSUE THREAD

**Issue by alexhambley** [eScienceLab/Cratey-Validator#5]
See here: https://flask-restx.readthedocs.io/en/latest/swagger.html

**Comment by alexhambley** [comment]
Requires a small refactor of post_routes.py. Will investigate next week (w/c 3rd Feb)

**Comment by alexhambley** [comment]
Should revisit this in this sprint, would require some refactoring but good practice to do so

**Comment by stain** [comment]
Can give automated docs for validator service, but how endpoints are defined has changed slightly.

**Comment by alexhambley** [comment]
Check with @stain + @douglowe on 1/4/25 to see if this is useful now (as we're decoupling products)

**Comment by ianhinder** [comment]
Agreed that this is low priority, and won't be addressed in the next sprint.

**Comment by douglowe** [comment]
switching to APIFlask library.

I've created a PR for this, with basic Swagger interface for the existing API calls.
